crengine/src/lvtinydom.cpp

```diff
@@ -17553,21 +17553,69 @@ bool ldomNode::getNodeListMarker( int & counterValue, lString32 & marker, int &
     case css_lst_upper_roman:
     case css_lst_lower_alpha:
     case css_lst_upper_alpha:
-        if ( counterValue<=0 ) {
-            // calculate counter
+        do {
+            // If this element has a valid value then use it avoiding a walk.
+            lString32 el_value = getAttributeValue(attr_value);
+            if ( !el_value.empty() ) {
+                int el_ivalue;
+                if ( el_value.atoi(el_ivalue) ) {
+                    counterValue = el_ivalue;
+                    break;
+                }
+            }
+
             // The UL > LI parent-child chain may have had some of our Boxing elements inserted
             ldomNode * parent = getUnboxedParent();
-            counterValue = 0;
-            // See if parent has a 'start' attribute that overrides this 0
+
+            // See if parent has a 'reversed' attribute.
+            int increment = parent->getAttributeValue(attr_reversed).empty() ? +1 : -1;
```

Comment:
A bit late, but:
https://html.spec.whatwg.org/multipage/grouping-content.html#attr-ol-reversed
> The reversed attribute is a boolean attribute. If present, it indicates that the list is a descending list (..., 3, 2, 1). If the attribute is omitted, the list is an ascending list (1, 2, 3, ...).
A number of attributes are boolean attributes. The presence of a boolean attribute on an element represents the true value, and the absence of the attribute represents the false value.

So, it seems `<ol reversed>` is valid - and your `parent->getAttributeValue(attr_reversed).empty() ` will match and you won't get things reversed. You could use: `parent->hasAttribute(attr_reversed) `
I don't know if `<ol reversed=false>` means it's not reversed and if more checks are needed.

(With all the _past HTML specs and what it meant at the time_ uncertainties :)

Reply:
Thanks fixed in https://github.com/buggins/coolreader/pull/227

Tested a few web browsers and they appear to reverse the list irrespective of the value of the attribute, even if 'false'.